### DISCUSSION TITLE
Support for textfsm parsing in fortinet_fortios

### DISCUSSION THREAD

**Discussion by yone2ks** [scrapli/scrapli_community#168] [General]
I believe that fortinet_fortios does not support textfsm parsing due to its based on GenericDriver. 
I would like to request support for textfsm parsing in fortinet_fortios. Additionally, there are fortinet templates available in ntc-templates.

Is it feasible to modify fortinet_fortios to support textfsm parsing?

**Comment by carlmontanari** [comment]
I think the answer is "no", but its also pretty trivial for you to just parse it yourself (or you could patch the driver object yourself too to make it even more "native" but that seems not worth the effort).

you can [get a template](https://github.com/carlmontanari/scrapli/blob/main/scrapli/helper.py#L25) w/ scrapli helper and [parse](https://github.com/carlmontanari/scrapli/blob/main/scrapli/helper.py#L91) too, or again, just do it yourself.

**Reply by yone2ks** [reply]
Thank you for your reply.
Understood. I plan to handle it as follows:
```
response = conn.send_command("some command")
response.textfsm_platform = 'fortinet'
structured_result = response.textfsm_parse_output()
```
personally, I think it wouldn't hurt to have the textfsm_platform variable in GenericDriver.
Is there any architectural reason within Scrapli why not having the textfsm_platform variable in GenericDriver?

**Comment by yone2ks** [comment]
This discussiion is closed because the first  discussion has been resolved.
